portfolio/src/main/java/com/google/sps/servlets/deleteData.java

```diff
@@ -20,7 +21,12 @@
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         Query query = new Query("Task");
-        long id = Long.parseLong(request.getParameter("id"));
+        long id = 0;
+        try {
+            id = Long.parseLong(request.getParameter("id"));
+        } catch (NumberFormatException e){
+            System.out.println("400 Error");
```

Comment:
The thing to do here is to set an error response code on the `response` and then `return`.

I'm not sure whether it's correct to do this with `response.sendError(400, "some useful error message")`, or with `response.setStatus(400)` plus setting the error message some other way. You may need to try it both ways and see which works.

You can test by running `curl http://colemanliyah-step-2020.appspot.com/deleteData?id=asdf -X post`, and also looking at the server logs to see if the server crashed or something.